ui/pages/confirmations/components/confirm/info/hooks/useUniswapShieldInfo.ts

```diff
@@ -0,0 +1,222 @@
+
+  useEffect(() => {
+    updateTransactionEventFragment(
+      {
+        properties: {
+          uniswap_shield: 'loading',
+        },
+      },
+      transactionId,
+    );
+
+    let parsedTransactionData: TransactionDescription;
+
+    try {
+      parsedTransactionData = contractInterface.parseTransaction({
+        data: data as Hex,
+      });
+    } catch (error) {
+      return;
+    }
+
+    const commands = parsedTransactionData.args.commands as string;
+    const inputs = parsedTransactionData.args.inputs as string[];
+    const commandBytes = commands.slice(2).match(/.{1,2}/gu) as string[];
+
+    let quotesInput;
+    let amountMin;
+
+    const sweepIndex = commandBytes.findIndex(
+      (commandByte) => commandByte === '04',
+    );
+
+    if (sweepIndex >= 0) {
+      const words = getWordsFromInput(inputs[sweepIndex]);
+      amountMin = wordToAmount(words[2]);
+      quotesInput = {
+        walletAddress: wordToAddress(words[1]),
+        srcChainId: chainId,
+        destChainId: chainId,
+        srcTokenAddress: getNativeTokenAddress(chainId),
+        destTokenAddress: wordToAddress(words[0]),
+        srcTokenAmount: amount ?? '0x0',
+        gasIncluded: false,
+        gasIncluded7702: false,
+      };
+    } else {
+      const seaportIndex = commandBytes.findIndex(
+        (commandByte) => commandByte === '10',
+      );
+
+      if (seaportIndex >= 0) {
+        const words = getWordsFromInput(inputs[seaportIndex]);
+        amountMin = wordToAmount(words[13]);
+        quotesInput = {
+          walletAddress: wordToAddress(words[28]),
+          srcChainId: chainId,
+          destChainId: chainId,
+          srcTokenAddress: wordToAddress(words[10]),
+          destTokenAddress: wordToAddress(words[16]),
+          srcTokenAmount: wordToAmount(words[12]),
+          gasIncluded: false,
+          gasIncluded7702: false,
+        };
+      }
+    }
+
+    if (!quotesInput || !amountMin) {
+      return;
+    }
+
+    fetchQuotes(quotesInput).then((quotes) => {
+      let selectedQuoteIndex = -1;
+      let highestMinDestTokenAmount = '0';
+
+      quotes.forEach((quote, index) => {
+        if (
+          new Numeric(quote.quote.minDestTokenAmount, 10).greaterThan(
+            new Numeric(highestMinDestTokenAmount, 10),
+          )
+        ) {
+          highestMinDestTokenAmount = quote.quote.minDestTokenAmount;
+          selectedQuoteIndex = index;
+        }
+      });
+
+      const percentageChangeInTokenAmount = new Numeric(
+        quotes[selectedQuoteIndex].quote.destTokenAmount,
+        10,
+      )
+        .minus(new Numeric(amountMin, 10))
+        .divide(new Numeric(amountMin, 10))
+        .times(100, 10)
+        .toNumber()
+        .toFixed(2);
+
+      const percentageChangeInTokenMinAmount = new Numeric(
+        quotes[selectedQuoteIndex].quote.minDestTokenAmount,
+        10,
+      )
+        .minus(new Numeric(amountMin, 10))
```

Comment:
Is it best we provide `slippage` in our requests to `fetchQuotes` to be explicit?

Maybe we could control via feature flag?

As that will influence our minimums.

Reply:
If we don't pass a slippage, does it uses the our swaps defaults? 
I believe they will be in a better position to define the ideal slippage than us as ideal slippage varies based on the token pair being swapped. 
A feature flag for this doesn't seem ideal as we would have to accomodate logic for specific token pairs there as well.